### DISCUSSION TITLE
Support for custom stylesheet

### DISCUSSION THREAD

**Discussion by kempf12358D** [hallvard/plantuml#158] [Q&A]
Is there a way to use a custom stylesheet with the eclipse plug-in? 

I was hoping to leverage something like this https://gist.github.com/jerieljan/4c82515ff5f2b2e4dd5122d354a82b7e#file-stylesheet-iuml as the default yellow boxes are meh.

**Comment by hallvard** [comment]
The plugin does support include directives, which is what I expect is used for this. Could you explain more precisely what kind of mechanism you envision?

**Comment by kempf12358D** [comment]
Having support for include directives is good. I just don't know how to incorporate an include in the automatic generation case.

Let me try and explain my situation in more detail. I am using Eclipse for Java development, have installed the plugin, and can generate UML diagrams in two ways; 
(1) I can generate a class diagram by selecting/highlighting a class or its file. I call the automatic generation.
(2) I can manually write plantuml code in the comments of a file.

It's easy to add include directives for case (2), but I would like to be able to setup a default include to use for the automatic generation (1). I just don't see a mechanism to do this. There is a Diagram Generation settings folder in the Preferences, but I don't know what type of file to add to this folder to affect the automatic gen mode.

The plugin is really useful. Thank you for creating this.  
-Jim

**Comment by hallvard** [comment]
A more general mechanism for injecting extra diagram text will be supported in the next release. You will be able to match on the path of the source file, the diagram generator, the kind of editor or view from which the diagram is generated and the diagram text itself, and then inject the desired extra diagram text.
